### PR TITLE
Allow Atlantis to Assume Roles

### PR DESCRIPTION
**Pull request by osterman**
## what
* Provide a role which is allowed to be assumed by pods running on a kops cluster

## why
* Enable GitOps workflows

### CODE REVIEW

**Comment by osterman** [comment]
This is tested & in use

**Review by solairerove — APPROVED** [review]

File: README.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    description: "Terraform module to provision an S3 bucket for Helm chart repository, and an IAM role and policy with permissions for Kops nodes to access the bucket"
+    url: "https://github.com/cloudposse/terraform-aws-kops-chart-repo"
+
+# Short description of this project
+description: |-
+  Terraform module to provision an IAM role for `atlantis` running in a Kops cluster, and attach an IAM policy to the role with permissions to modify infrastructure.
+
+  ## Overview
+
+  This module assumes you are running [atlantis](https://github.com/kubernetes-incubator/atlantis) in a Kops cluster. We recommend using it together with [`kiam`](https://github.com/uswitch/kiam) to permit pods to assume roles.
```
Comment (aknysh) [inline README.yaml]: wrong link
https://github.com/kubernetes-incubator/atlantis

File: README.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  The module uses [terraform-aws-kops-metadata](https://github.com/cloudposse/terraform-aws-kops-metadata) to lookup resources within a Kops cluster for easier integration with Terraform.
+
+
+# How to use this project
+usage: |-
+  ```hcl
+  module "kops_atlantis" {
+    source       = "git::https://github.com/cloudposse/terraform-aws-kops-atlantis.git?ref=master"
+    namespace    = "cp"
```
Comment (aknysh) [inline README.yaml]: we decided to use `eg` for the namespace
namespace    = "eg"

File: README.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    stage        = "prod"
+    name         = "atlantis"
+    cluster_name = "us-east-1.cloudposse.com"
```
Comment (aknysh) [inline README.yaml]: `cloudposse.co`, since all our docs use that domain

File: README.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    masters_name = "masters"
+    nodes_name   = "nodes"  
+
+    tags = {
+      Cluster = "us-east-1.cloudposse.com"
```
Comment (aknysh) [inline README.yaml]: cloudposse.co

File: README.yaml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+  }
+  ```
+
+include:
+  - "docs/targets.md"
+  - "docs/terraform.md"
+
+# Contributors to this project
+contributors:
```
Comment (aknysh) [inline README.yaml:113]: why the main contributor is missing? :)

```
contributors:
  - name: "Erik Osterman"
    homepage: "https://github.com/osterman"
    avatar: "http://s.gravatar.com/avatar/88c480d4f73b813904e00a5695a454cb?s=144"
    github: "osterman"
  - name: "Igor Rodionov"
    homepage: "https://github.com/goruha/"
    avatar: "http://s.gravatar.com/avatar/bc70834d32ed4517568a1feb0b9be7e2?s=144"
    github: "goruha"
  - name: "Andriy Knysh"
    homepage: "https://github.com/aknysh/"
    avatar: "https://avatars0.githubusercontent.com/u/7356997?v=4&u=ed9ce1c9151d552d985bdf5546772e14ef7ab617&s=144"
    github: "aknysh"
```

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags (e.g. map(`Cluster`,`us-east-1.cloudposse.com`)"
+}
+
+variable "cluster_name" {
+  type        = "string"
+  description = "Kops cluster name (e.g. `us-east-1.cloudposse.com` or `cluster-1.cloudposse.com`)"
```
Comment (aknysh) [inline variables.tf]: cloudposse.co

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+variable "attributes" {
+  type        = "list"
+  default     = []
+  description = "Additional attributes (e.g. `1`)"
+}
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags (e.g. map(`Cluster`,`us-east-1.cloudposse.com`)"
```
Comment (aknysh) [inline variables.tf]: cloudposse.co

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,56 @@
+variable "namespace" {
+  type        = "string"
+  description = "Namespace (e.g. `cp` or `cloudposse`)"
```
Comment (aknysh) [inline variables.tf]: (e.g. `eg` or `cp`)

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    principals {
+      type = "AWS"
+
+      identifiers = [
+        "${module.kops_metadata.masters_role_arn}",
```
Comment (aknysh) [inline main.tf:44]: why masters? or only masters?
do we want just nodes to assume the role?

**Review by aknysh — COMMENTED** [review]
LGTM, see comments

File: main.tf (hunk 9 above)
Comment (osterman) [reply main.tf:44]: with `kiam` you *only* want to grant access to masters.

**Review by aknysh — APPROVED** [review]

**Pull request merged**